File: src/main/java/com/partymakers/shareparty/domain/friends/usecase/impl/RegisterFriendImpl.java
package com.partymakers.shareparty.domain.friends.usecase.impl;

import com.partymakers.shareparty.domain.friends.entity.Friend;
import com.partymakers.shareparty.domain.friends.port.FriendsRepository;
import com.partymakers.shareparty.domain.friends.usecase.RegisterFriend;
import com.partymakers.shareparty.domain.party.usecase.exception.AlreadyExistException;
import com.partymakers.shareparty.domain.party.usecase.exception.NotFoundException;

import org.springframework.data.repository.CrudRepository;

import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class RegisterFriendImpl implements RegisterFriend {
    private final FriendsRepository repository;

    @Override
    public void registerFriend(Friend friend) {
        repository.findOneById(friend.getNickName()).ifPresentOrElse(
            foundFriend ->{throw new AlreadyExistException("Friend " + friend.getName() + " already exist");},
            () -> repository.save(friend));
    }
}

File: src/main/kotlin/com/partymakers/shareparty/domain/friends/usecase/RegisterFriendImpl.kt
package com.partymakers.shareparty.domain.friends.usecase

import com.partymakers.shareparty.domain.friends.entity.Friend
import com.partymakers.shareparty.domain.friends.port.FriendsRepository
import com.partymakers.shareparty.domain.party.usecase.exception.AlreadyExistException
import org.springframework.stereotype.Service

@Service
class RegisterFriendImpl(
    private val repository: FriendsRepository
) : RegisterFriend {

    override fun registerFriend(friend: Friend) {
        repository.findOneById(friend.nickName)
            ?.let { foundFriend -> throw AlreadyExistException("Friend " + friend.name + " already exist") }
            ?: run { repository.save(friend) }
    }

}